src/snapred/backend/dao/indexing/Versioning.py

```diff
@@ -0,0 +1,61 @@
+from typing import Any, Optional
+
+from numpy import integer
+from pydantic import BaseModel, computed_field, field_serializer
+from snapred.meta.Config import Config
+
+VERSION_START = Config["version.start"]
+VERSION_NONE_NAME = Config["version.error"]
+VERSION_DEFAULT_NAME = Config["version.default"]
+VERSION_DEFAULT = -1  # SNAPRed Internal flag for default version
+
+
+class VersionedObject(BaseModel):
+    __version: Optional[int] = None
+
+    def __init__(self, **kwargs):
+        version = kwargs.pop("version", None)
+
+        if version == VERSION_NONE_NAME:
+            version = None
+        elif version == VERSION_DEFAULT_NAME:
+            version = VERSION_DEFAULT
+        elif version == VERSION_DEFAULT or version is None:
+            pass
+        elif isinstance(version, integer):
+            version = int(version)
+        elif not isinstance(version, int) or version < VERSION_START:
+            raise ValueError(f"Cannot initialize version as {version}")
+        super().__init__(**kwargs)
+        self.__version = version
```

Comment:
I feel as through parsing/translation would benefit from being seated in their own methods.

Reply:
I agree.  Let me see what I can do